data/DataHandler.php

```diff
@@ -0,0 +1,247 @@
+<?php
+//==================================
+// Class Definition of DataHandler
+//==================================
+require_once __DIR__.'/../utility/Equipment.php';
+require_once __DIR__.'/../utility/Room.php';
+session_start();
+
+
+class DataHandler {
+    
+    protected const MAX_SPACE = 1000;
+    protected const WAREHOUSE = 'warehouse';
+    protected const NONE = "";
+
+    private string $rooms = 'room_list', $equipments = 'equip_list';
+
+    /**
+     * Constructor for the Handler.
+     * Initializes two arrays containing all equipment and all rooms.
+     * Automatically creates the warehouse.
+     */
+    public function __construct() {
+        $_SESSION[$this->rooms]  ?? $_SESSION[$this->rooms] = [self::WAREHOUSE => new Room(self::WAREHOUSE, self::MAX_SPACE)];
+        $_SESSION[$this->equipments] ?? $_SESSION[$this->equipments] = [];
+        
+    }
+    
+    /************** Equipment Methods ****************/
+    /**
+     * Adds an equipment to the database. Returns false if equipment already exists.
+     * If overwrite flag is set to true, the old equipment will be overwritten.
+     * @param  Equipment $equip to be added
+     * @param  string $room Optional room location. 
+     * @param  bool $overwrite Default will not allow overwriting, if set to true, overwriting will be allowed.
+     * @return bool
+     */
+    public function add_equipment(Equipment $new_equip, String $room = self::NONE, bool $overwrite = false) : bool {
+        if (!isset($new_equip)) {
```

Comment:
The signature of the function forces this var to be set. Did u mean to check if it is empty somehow. i.e. room without values?
Remember - separation of concerns. This class is responsible to write into the db and read.
You have another class that is supposed to make sure the input/ouput is valid.
It is legal, if u wish, to use the validator inside the data handler class. But, I should not see validation logic in the data Nadler

Reply:
The function allows users to add an equipment directly to a room while adding to the database. However, if the equipment is meant to stay in the inventory, then no location will be specified and room/equipment location will automatically initialize to NONE